fix: use language specified in SiteConfiguration

Agrendalath opened the pull request:
## Context
This feature was implemented in #27696 to replace the session's language in the request.
#28796 moved the process from the request to the response, which made this feature unusable (because the request was already processed). #28796 also made this feature set the language cookie. However, it is overwritten by user preferences.

## Description

To fix the described problem, this ignores user preferences and alters the cookie of a request instead of setting the cookie in the response. It also moves this feature from `dark_lang` to `lang_pref` middleware. It fits better here because it directly overwrites the language from user preferences.

Additionally, we are removing setting the session language by this feature. As mentioned in [the release notes](https://docs.djangoproject.com/en/3.2/releases/3.0/#miscellaneous), Django 3.2 no longer checks it in the `LocalMiddleware`. You can find the related code [here](https://github.com/django/django/blob/stable/3.2.x/django/utils/translation/trans_real.py#L530).

## Testing instructions

1. Visit https://pr31408.sandbox.opencraft.hosting/dashboard. The page should be in English.
2. Visit https://fr.pr31408.sandbox.opencraft.hosting/dashboard (ignore the SSL warning). The page should be in French.
3. Visit https://pl.pr31408.sandbox.opencraft.hosting/dashboard (ignore the SSL warning). The page should be in Polish.
4. On any of the Sites, go to the [/update_lang/ page](https://fr.pr31408.sandbox.opencraft.hosting/update_lang/) and select a different language (e.g., `de`). This should change the language of the page. It does not affect the "Started - Feb 5, 2013" text on the dashboard, but this is an upstream bug (not related to this PR).

Note: if you would like to test this locally, you should add `SITE_CACHE_TTL = 0` to your `lms/envs/private.py`. Otherwise, you will need to wait for the SiteConfiguration cache to be invalidated.

## Deadline

"None"

## Other information
Alternatively, we could overwrite the cookie of the response after it's set from user preferences. However, it is not an ideal solution because when users switch between Sites with different languages, the first response will use the language of the previous page.

Sandbox: https://pr31408.sandbox.opencraft.hosting/
Private-ref: [BB-6930](https://tasks.opencraft.com/browse/BB-6930)

Agrendalath committed:
fix: use language specified in SiteConfiguration

This feature was implemented in b01544d690 to replace the session's language
in the request. 44ddbdf925 moved the process from the request to the response,
which made this feature unusable (because the request was already processed).
44ddbdf925 also made this feature set the language cookie. However, it is
overwritten by user preferences.
To fix this, we could overwrite the cookie of the response after it's set from
user preferences. However, it is not an ideal solution because when users
switch between Sites with different languages, the first response will use the
language of the previous page. Therefore, this ignores user preferences and
alters the cookie of a request instead.

openedx-webhooks commented:
Thanks for the pull request, @Agrendalath!

As a core committer in this repo, you can merge this once the pull request is approved per the [core committer reviewer requirements](https://openedx.atlassian.net/wiki/spaces/COMM/pages/1529675973/Open+edX+Core+Committers#Reviews-and-Ownership) and according to the agreement with your edX Champion.



<!-- comment:welcome-core-committer -->
<!-- jenkins ok to test -->
<!-- data: eyJkcmFmdCI6IGZhbHNlfQ== -->

Agrendalath committed:
test: fix `TestCourseStatusGET`

This test stopped throwing the `TransactionManagementError` once we added the
`site_configuration.get_value()` call to the language preferences middleware.

Agrendalath commented on lms/djangoapps/mobile_api/users/tests.py:
```diff
@@ -501,21 +501,17 @@ def test_success_v1(self):
-        try:
-            with transaction.atomic():
-                self.url_method(url, data=data, **kwargs)
-                assert False
```
This test does not seem to be related, but it stopped throwing the `TransactionManagementError` once we added the `site_configuration.get_value()` call to the language preferences middleware.

This was not the case before because this feature was gated behind `DarkLangConfig`, which is disabled in this test. This gating does not make much sense, as this feature does not depend on  `DarkLang`.

pkulkark replied:
Nice find @Agrendalath!

pkulkark reviewed (approved):
LGTM :+1:

- [x] I tested this: verified that the language set in SiteConfiguration is used as described in testing instructions
- [x] I read through the code
- [ ] ~I checked for accessibility issues~ N/A
- [ ] ~Includes documentation~ N/A

openedx-webhooks commented:
@Agrendalath 🎉 Your pull request was merged!  Please take a moment  to [answer a two question survey](https://docs.google.com/forms/d/e/1FAIpQLSceJOyGJ6JOzfy6lyR3T7EW_71OWUnNQXp68Fymsk3MkNoSDg/viewform?usp=pp_url&entry.1671973413=openedx/edx-platform&entry.867055334=https://github.com/openedx/edx-platform/pull/31408&entry.1484655318=https://github.com/Agrendalath&entry.752974735=2022-12-06+16:23&entry.1917517419=2022-12-09+16:20&entry.2133058324=Yes) so we can improve your experience in the future.

<!-- comment:end_survey -->
